internal/blocksync/reactor.go

```diff
@@ -554,6 +442,158 @@ FOR_LOOP:
 	}
 }
 
+// setupDeferred sets up deferred calls for poolRoutine.
+func (bcR *Reactor) setupDeferred(trySyncTicker, statusUpdateTicker, switchToConsensusTicker *time.Ticker) {
+	bcR.metrics.Syncing.Set(0)
+	trySyncTicker.Stop()
+	statusUpdateTicker.Stop()
+	switchToConsensusTicker.Stop()
+}
+
+// setupPoolRoutine initializes the variables used in the poolRoutine.
+func (bcR *Reactor) setupPoolRoutine() (chan struct{}, *time.Ticker, *time.Ticker, *time.Ticker, uint64, string, sm.State, time.Time, float64, bool) {
+	bcR.metrics.Syncing.Set(1)
+
+	trySyncTicker := time.NewTicker(trySyncIntervalMS * time.Millisecond)
+	statusUpdateTicker := time.NewTicker(statusUpdateIntervalSeconds * time.Second)
+
+	if bcR.switchToConsensusMs == 0 {
+		bcR.switchToConsensusMs = switchToConsensusIntervalSeconds * 1000
+	}
+	switchToConsensusTicker := time.NewTicker(time.Duration(bcR.switchToConsensusMs) * time.Millisecond)
+
+	blocksSynced := uint64(0)
+
+	chainID := bcR.initialState.ChainID
+	state := bcR.initialState
+
+	lastHundred := time.Now()
+	lastRate := 0.0
+
+	didProcessCh := make(chan struct{}, 1)
+
+	initialCommitHasExtensions := (bcR.initialState.LastBlockHeight > 0 && bcR.store.LoadBlockExtendedCommit(bcR.initialState.LastBlockHeight) != nil)
+
+	return didProcessCh, trySyncTicker, statusUpdateTicker, switchToConsensusTicker, blocksSynced, chainID, state, lastHundred, lastRate, initialCommitHasExtensions
+}
+
+// handlePeerError processes an error received from a peer.
+// If the peer that caused the error is still connected, it stops the peer and logs the error.
+// This function is used in the poolRoutine of the Reactor to handle errors received from peers.
```

Comment:
```suggestion
```

don't think we need that (the place where we call this is in the same file, so it's easy to locate)